packages/apps/plugins/plugin-markdown/src/MarkdownPlugin.tsx

```diff
@@ -243,6 +245,19 @@ export const MarkdownPlugin = (): PluginDefinition<MarkdownPluginProvides> => {
             return getCustomExtensions(doc /*, query */);
           }, [doc, space, settings.values.editorMode]);
 
+          const fileManagerPlugin = useResolvePlugin(parseFileManagerPlugin);
+          const onFileUpload = async (file: File) => {
+            if (space === undefined) {
+              throw new Error('Unable to determine space used');
+            }
+
+            if (fileManagerPlugin?.provides.file.upload === undefined) {
+              throw new Error('No upload method was provided');
+            }
+
+            return await fileManagerPlugin.provides.file.upload(file, space);
+          };
```

Comment:
Can you convert this to a `useCallback` with `space` and `fileManagerPlugin` as deps?

Reply:
I also think that it would be good to omit `onFileUpload` if the plugin doesn't exist rather than erroring when its called. This would give the editor a hint to disable/hide the functionality if it's not available.

Reply:
Done 👍  I removed the errors @wittjosiah, now just returns `undefined` if the space and file manager plugin aren't present. Unless you want the `onFileUpload` variable itself to be undefined if those two dependencies aren't present?

Reply:
Yeah the `onFileUpload` prop should be undefined if the file manager plugin isn't present.

Reply:
Yeah, what Zaymon said. Because the editor handles the case where `onFileUpload` isn't present by hiding those controls, if the capabilities are not available the callback itself should just be `undefined`. Once that change is made, I think we should be able to merge this.

Reply:
Updated ✅